python/cuml/dask/ensemble/base.py

```diff
@@ -123,7 +125,7 @@ def _concat_treelite_models(self):
         bytes format.
         """
         model_serialized_futures = list()
-        for w in self.workers:
+        for w in self.active_workers:

```

Comment:
The `DistributedDataHandler` has a `workers` instance variable that you can can use to save a few lines of code here. You should be able to set `self.active_workers = data.workers` right after the call to `DIstributedDataHandler.create()`.

Reply:
IMO, we should eventually hold off on calling `self._create_models()` until `fit()`. That would allow us to only create models on workers that hosted training data and would remove the need for tracking a separate instance variable for the `active_workers`. This would also fix @JohnZed's concern about the need to update parameters when `self.active_workers != self.workers`. Is there any compelling reason to call `_create_models()` in the constructor, rather than during training?

Reply:
I'm not sure I understand how moving `self._create_models()` in `fit()` would solve this issue?

It seems to be coming from the fact that `DistributedDataHandler` is not splitting the data into all workers correctly, and even having it just before data is split might still result in this problem, I think. I might be missing something here!

Reply:
Nvm, John explained it. I think we can do that after this release.